fedimint-client/src/module/mod.rs

```diff
@@ -63,6 +64,27 @@ pub trait ClientModule: Debug + MaybeSend + MaybeSync + 'static {
         &self,
         output: &<Self::Common as ModuleCommon>::Output,
     ) -> TransactionItemAmount;
+
+    fn supports_backup(&self) -> bool {
+        false
+    }
+
+    async fn backup(
+        &self,
+        _dbtx: &mut ModuleDatabaseTransaction<'_>,
```

Comment:
@elsirion 

Does this seem reasonable?

To be able to generically call every module to make a backup, I need to have it as a unified method on the common trait. But then them modules don't seem to have the `db` inside themselves and seem to rely on `*Ext` traits to grab it from the `Client::inner`. So I guess I'll just have to pass it?

A bit unclear to me how to actually get `ModuleDatabaseTransaction<'_>`, but maybe I'm just tired and missed it.

Reply:
I don't understand if there's a reason we can't just pass modules api and their isolated db handle in the constructor. This way we wouldn't need `*Ext` traits.

Reply:
@elsirion  Hmm... I don't think extension traits are worth it if they are there just to "get the default instance id for a given trait". That could be just a method.